test/functional/apps/discover/_saved_queries.ts

```diff
@@ -128,20 +129,22 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
         await savedQueryManagementComponent.loadSavedQuery('OkResponse');
         const timePickerValues = await PageObjects.timePicker.getTimeConfigAsAbsoluteTimes();
         expect(await filterBar.hasFilter('extension.raw', 'jpg')).to.be(true);
-        expect(timePickerValues.start).to.not.eql(PageObjects.timePicker.defaultStartTime);
-        expect(timePickerValues.end).to.not.eql(PageObjects.timePicker.defaultEndTime);
+        expect(timePickerValues.start).to.eql(from);
+        expect(timePickerValues.end).to.eql(to);
       });
 
       it('preserves the currently loaded query when the page is reloaded', async () => {
         await browser.refresh();
         const timePickerValues = await PageObjects.timePicker.getTimeConfigAsAbsoluteTimes();
         expect(await filterBar.hasFilter('extension.raw', 'jpg')).to.be(true);
-        expect(timePickerValues.start).to.not.eql(PageObjects.timePicker.defaultStartTime);
-        expect(timePickerValues.end).to.not.eql(PageObjects.timePicker.defaultEndTime);
-        await retry.waitFor(
-          'the right hit count',
-          async () => (await PageObjects.discover.getHitCount()) === '2,792'
-        );
+        expect(timePickerValues.start).to.eql(from);
+        expect(timePickerValues.end).to.eql(to);
+        await retry.waitForWithTimeout('the right hit count', 65000, async () => {
+          const hitCount = await PageObjects.discover.getHitCount();
+          log.debug(`Found hit count is ${hitCount}. Looking for 2,792.`);
+          await PageObjects.common.sleep(5000);
```

Comment:
Is there a specific condition that we can wait for with using `waitFor` or `waitForWithTimeout` rather than a hard coded sleep? Just to reduce the chance of flakiness.

Reply:
I don't think that this solves the flakiness, I started flaky test runner with more runs, and got several errors: https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/350

Reply:
looking at the logs I can see a suspicious line there 

```
[00:07:56]           │ debg ... sleep(501) start
[00:07:56]           │ debg browser[INFO] http://localhost:5620/51519/bundles/core/core.entry.js 7:149182 "Detected an unhandled Promise rejection.
[00:07:56]           │      Error: EsError: it is mandatory to set the [nested] context on the nested sort field: [nested.timestamp]."
[00:07:56]           │ warn browser[SEVERE] http://localhost:5620/51519/bundles/plugin/data/kibana/data.plugin.js 0:313795 Uncaught es_error_EsError: EsError: it is mandatory to set the [nested] context on the nested sort field: [nested.timestamp].
[00:07:56]           │ debg ... sleep(501) end
```

Reply:
Maybe, seems it seems to happen before the test is even executed, it's some sort of leftover from a previous test?

Reply:
I added a few intermediate verifications of the Discover hit count in the before method.  And I also realized that there was a call to `setQuery` but not a call to `submitQuery`.  I wouldn't think that was intentional, so I added the call to `submitQuery`.
I started a new flaky test runner job of 100 runs here https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/358

Reply:
My latest changes seem to have resolved the flakiness in the first describe block but is now still failing 6/100 runs on the first test in the second describe block.  Why on 6/100 runs are there no documents here? 

discover app saved queries saved objects saved query management component functionality "before all" hook for "should show the saved query management component when there are no saved queries"

![image](https://user-images.githubusercontent.com/13542669/161109284-47655348-9ad2-4274-800f-d734aaeb21bb.png)

Reply:
```TypeScript
       await PageObjects.discover.selectIndexPattern('date-nested');

        expect(await filterBar.hasFilter('extension.raw', 'jpg')).to.be(false);
        expect(await queryBar.getQueryString()).to.eql('');

        await PageObjects.discover.selectIndexPattern('logstash-*');

        expect(await filterBar.hasFilter('extension.raw', 'jpg')).to.be(false);
        expect(await queryBar.getQueryString()).to.eql('');

        // reset state
        await savedQueryManagementComponent.deleteSavedQuery('test-unselect-saved-query');
      });
    });

    describe('saved query management component functionality', function () {
      before(async () => await setUpQueriesWithFilters());
```    
In the given screenshot `date-nested` is selected ..., but according to the code `logstash-*` should be selected... so I think the flakiness is a result of the previous test not switching to `logstash-*`

Reply:
Yep.  That's where I'm debugging right now.  I'm adding this code when it `selectIndexPatthen('logstash-*')`
```
        await PageObjects.discover.selectIndexPattern('logstash-*');
        const currentDataView = await PageObjects.discover.getCurrentlySelectedDataView();
        expect(currentDataView).to.be('logstash-*');
        await retry.try(async function tryingForTime() {
          const hitCount = await PageObjects.discover.getHitCount();
          expect(hitCount).to.be('4,731');
        });
```

I'm not sure if this will fix the flakiness but if it fails it might give more clues.